### PR TITLE
Add -c|--cd to specify container's initial working directory.

### LINKED ISSUE #72: Allow to specify initial working directory in container
Especially when `--no-home` is not passed, the user might expect to end up in his / her home directory. 
In general, it would be a nice shortcut to be able to specify the IWD in the container.

### PR DESCRIPTION
**Pull request by olifre**
Default remains /.

This fixes #72 .

Signed-off-by: Oliver Freyermuth <o.freyermuth@googlemail.com>

### CODE REVIEW

File: bin/ch-run.c [hunk 1]
```diff
@@ -266,6 +269,10 @@ void enter_udss(char * newroot, bool writable, struct bind * binds,
 #ifdef SETUID
    privs_drop_temporarily();
 #endif
+
+   if (args.initial_working_dir != NULL) {
+      TRY (chdir(args.initial_working_dir));
+   }
```
Comment (reidpr) [inline bin/ch-run.c]: House style is currently no braces here.

File: bin/ch-run.c [hunk 2]
```diff
@@ -314,6 +321,9 @@ static error_t parse_opt(int key, char * arg, struct argp_state * state)
    case -2:
       as->private_home = true;
    break;
```
Comment (reidpr) [inline bin/ch-run.c]: While you're in there, would you indent this `break` to match?

**Comment by reidpr** [comment]
Looks nice, thank you.

Would you add a test and then rebase into a single commit?

Off the top of my head I can think of testing that `--cd` missing ends up at `/` and present ends up where requested.

File: bin/ch-run.c (hunk 2 above)
Comment (olifre) [reply bin/ch-run.c]: Done as part of the stashed commit.

**Comment by olifre** [comment]
All done, let's see whether travis confirms my local testing. 
I added also a third test: Check that we really get an error if trying to `--cd` to a directory which does not exist in the container.

**Comment by reidpr** [comment]
Merged, thank you!

I did add a couple of commits to make the error friendlier (4f959b5) and to make the tests a bit more idiomatic (f430ecb).

**Pull request merged**